R/ops-addition.r

```diff
@@ -43,33 +43,28 @@ add_period_to_period <- function(per2, per1) {
 
 add_period_to_date <- function(per, date) {
 
-  lt <- as.POSIXlt(date)
+  new <- update(date,
+                years = year(date) + per@year,
+                months = month(date) + per@month)
 
-  ## add months and years with no backwards rollover
-  ms <- month(per) + year(per) * 12
-  lt <- add_months(lt, ms)
+  new <- solve_monthday_mismatches(new, mday(date))
 
-  new <- update(lt,
-                days = mday(lt) + per@day,
-                hours = hour(lt) + per@hour,
-                minutes = minute(lt) + per@minute,
-                seconds = second(lt) + per@.Data)
+  new <- update(new,
+                days = mday(date) + per@day,
+                hours = hour(date) + per@hour,
+                minutes = minute(date) + per@minute,
+                seconds = second(date) + per@.Data)
 
-  if (is.Date(date) && sum(new$sec, new$min, new$hour, na.rm = TRUE) != 0)
-    return(new)
+  if (is.Date(date)) return(new)
```

Comment:
So you are always returning datetime now?

Reply:
Thanks for the feedback! The rationale was that update.Date already has a check on zero h/m/s (also sufering from the NA issue so should be rewritten as well). By avoiding the conversion to POSIXlt, update.Date would be used for Dates instead of update.POSIXlt. But you are a right about the performance. There is the option to combine both worlds though. Keep the conversion to POSIXlt with the years/months addition, then have an if for Dates that converts back to a Date and calls update.Date. Microbenchmark showed this was faster for Dates then the current implementation. What do you think?

Reply:
Yes, that's a good idea. I think you can even avoid the conversion back to date if the if-else is big enough with an update on the `date` argument directly, but let's not bother with stylistics for now. All of these code will go away once we move on top of time-change. 

Please note that I might not merge this PR before the next release (very urgent) as people might rely on POSIXlt output and I don't want to break stuff right now.